permissions/utils.py

```diff
@@ -0,0 +1,43 @@
+def check_permission(course, unit, user):
+    ''' Check if user has permissions for a given unit '''
+
+    teams = list(course.team_set.all())

```

Comment:
you don't need to typecast to list. Also do you need all the columns from team set?
You can use `values_list` to just get the ids and use it in the next filter.